package/src/services/check-parser/parser.ts

```diff
@@ -46,89 +56,110 @@ export class Parser {
+        return Parser.tryReadFileExt(filePath, [SupportedExtensions.JS])
+      } else {
+        return Parser.tryReadFileExt(filePath, [SupportedExtensions.TS, SupportedExtensions.JS])
+      }
     }
   }
 
-  static parseDependenciesForFile (filePath: string): { localDependencies: string[], npmDependencies: string[] } {
+  static tryReadFileExt (filePath: string, exts: Array<SupportedExtensions>) {
+    for (const extension of exts) {
+      try {
+        const fullPath = filePath + extension
+        const content = fs.readFileSync(fullPath, { encoding: 'utf-8' })
+        return { filePath: fullPath, content }
+      } catch (err) {}
+    }
+    throw new Error(`Cant find file ${filePath}`)
+  }
+
+  static parseDependencies (filePath: string, contents: string):
+  { localDependencies: string[], npmDependencies: string[], error?: any } {
     const localDependencies = new Set<string>()
     const npmDependencies = new Set<string>()
-    const contents = fs.readFileSync(filePath, { encoding: 'utf8' })
 
     const extension = path.extname(filePath)
-
-    if (extension === SupportedExtensions.JS) {
-      const ast = acorn.parse(contents, {
-        allowReturnOutsideFunction: true,
-        ecmaVersion: 'latest',
-        allowImportExportEverywhere: true,
-      })
-      walk.simple(ast, Parser.jsNodeVisitor(localDependencies, npmDependencies))
-    } else if (extension === SupportedExtensions.TS) {
-      const tsParser = getTsParser()
-      const ast = tsParser.parse(contents, {})
-      // The AST from typescript-estree is slightly different from the type used by acorn-walk.
-      // This doesn't actually cause problems (both are "ESTree's"), but we need to ignore type errors here.
-      // @ts-ignore
-      walk.simple(ast, Parser.tsNodeVisitor(tsParser, localDependencies, npmDependencies))
-    } else {
-      throw new Error(`Unsupported file extension for ${filePath}`)
+    try {
+      if (extension === SupportedExtensions.JS) {
+        const ast = acorn.parse(contents, {
+          allowReturnOutsideFunction: true,
+          ecmaVersion: 'latest',
+          allowImportExportEverywhere: true,
+        })
+        walk.simple(ast, Parser.jsNodeVisitor(localDependencies, npmDependencies))
+      } else if (extension === SupportedExtensions.TS) {
+        const tsParser = getTsParser()
+        const ast = tsParser.parse(contents, {})
+        // The AST from typescript-estree is slightly different from the type used by acorn-walk.
+        // This doesn't actually cause problems (both are "ESTree's"), but we need to ignore type errors here.
+        // @ts-ignore
+        walk.simple(ast, Parser.tsNodeVisitor(tsParser, localDependencies, npmDependencies))
+      } else {
+        throw new Error(`Unsupported file extension for ${filePath}`)
+      }
+    } catch (err) {
+      return {
+        localDependencies: Array.from(localDependencies),
+        npmDependencies: Array.from(npmDependencies),
+        error: err,
```

Comment:
Very go-lang style 😄

Reply:
Yeah. Error probe operation, like parsing, are really hard to achieve with the try/catch syntax